### ISSUE TITLE
Allow toggling of the ^ & % stuff within the rename function.

### ISSUE THREAD

**Issue by SteelTitanium** [sirjuddington/SLADE#646]
**SLADE Version**: (3.1.1.5)
**Operating System**: Windows
An option to allow use of characters like ^ & % when renaming a lump would be nice.

**Comment by tlm-2501** [comment]
From what I understand, lump names have to follow a very specific format (letters, numbers and very select special characters, such as underscores and the minus sign), and that isn't a SLADE limitation, but rather that of the DOOM engine itself.

**Comment by SteelTitanium** [comment]
Yes, I understand that this would lead to incompatibility with many existing source-ports, that's why it would be perfect as an setting you could turn on and off as you want, obviously warning the user that this may lead to incompatibility.

**Comment by Gaerzi** [comment]
There's no real limitation on lump names as far as the archive format is concerned; any string of eight ASCII characters is valid. It doesn't have the limitations of 8.3 DOS filenames.

The issue here is when renaming several lumps at once, then ?, *, ^, ^^, %, %%, &, and && are interpreted as special effect rather than their own characters. A possible option would be to add escape characters (so for instance \^ would be interpreted as the character ^, and \\ as the character \), another is to allow to just disable the renaming masks.

Note that when renaming a single lump (or using the "rename each" option instead of mass rename), the masks are already disabled, so you can perfectly well name a lump ^%%^^?*&. (Though to use carets you'll have to select the percent encoding in preferences->editing->general.)

**Comment by LordOZ98** [comment]
was trying to create a custom character for SRB2 and realized Slade no longer supports those, Huh ?
